### PR TITLE
[Fix #1725] --color option forces color output (even when not printing to TTY)

### LINKED ISSUE #1725: "force TTY" command line option
I'd like colorized output in Gitlab CI, but there "TTY only" color output in Rubocop. Maybe it'll be great to have "force TTY" command line option? What do you think?

### PR DESCRIPTION
**Pull request by alexdowad**

### CODE REVIEW

File: lib/rubocop/formatter/formatter_set.rb [hunk 1]
```diff
@@ -73,7 +76,7 @@ def add_formatter(formatter_type, output_path = nil)
           output = $stdout
         end
 
-        self << formatter_class.new(output)
+        self << formatter_class.new(output, @options)
```
Comment (jonas054) [inline lib/rubocop/formatter/formatter_set.rb]: This must be a problem for existing custom formatters whose constructors only take one argument.

Comment (alexdowad) [inline lib/rubocop/formatter/formatter_set.rb]: That is why the constructor for `BaseFormatter` is now declared as `def initialize(output, options = {})`.

Comment (jonas054) [inline lib/rubocop/formatter/formatter_set.rb]: But if a custom formatter overrides that with `def initialize(output)`, you still get a `wrong number of arguments (2 for 1)` error from this call.

Comment (alexdowad) [inline lib/rubocop/formatter/formatter_set.rb]: Oooh. You're right.

Comment (alexdowad) [inline lib/rubocop/formatter/formatter_set.rb]: Does RuboCop have a documented API for extension?

Comment (jonas054) [inline lib/rubocop/formatter/formatter_set.rb]: This is what we have: https://github.com/bbatsov/rubocop#extensions

Comment (alexdowad) [inline lib/rubocop/formatter/formatter_set.rb]: OK. Looks like changing the constructor signature is not viable. Let me come up with another way.

Comment (alexdowad) [inline lib/rubocop/formatter/formatter_set.rb]: BTW Do you have tests for the extension API?

Comment (jonas054) [inline lib/rubocop/formatter/formatter_set.rb]: Apparently not. :smile: If we had, they would have caught this thing.
Such tests would be a good idea, though.

Comment (bbatsov) [inline lib/rubocop/formatter/formatter_set.rb]: Btw, I'm not sure how many custom formatters exist in the wild, so changing this is not out of question. And we're not at 1.0 yet. :-)

Comment (jonas054) [inline lib/rubocop/formatter/formatter_set.rb]: Adding a setter in the base class so you can call `self.last.options = @options` here would work, I believe. You would have to move any logic depending on options out of the formatter constructors, and put it somewhere else.

**Comment by bbatsov** [comment]
Btw, what's the state here?

**Comment by alexdowad** [comment]
Closing.

**Pull request closed without merging**